[Stack Monitoring] update rules queries to support metricbeat 8.0

neptunian opened the pull request:
### Summary
resolves https://github.com/elastic/kibana/issues/125035.  Fixes queries in Stack Monitoring rules to be compatible with metricbeat 8 mappings.  Except for CCR, I tested these on `main` with metricbeat 8.1.0.  CCR i tested with metricbeat 8.1.0 mappings but used my own data as I couldn't get a local CCR setup.  

I tested these manually by looking at the [response](https://github.com/elastic/kibana/blob/8.0/x-pack/plugins/monitoring/server/lib/alerts/fetch_missing_monitoring_data.ts#L130) and the [data it returned](https://github.com/elastic/kibana/blob/8.0/x-pack/plugins/monitoring/server/lib/alerts/fetch_missing_monitoring_data.ts#L158).  Ideally we need a proper way to run functional tests on these rule queries.

### TODO
- [ ] test CCR with real metricbeat 8.0.0+ data
- [ ] test all with internal collection data
- [ ] populate the `elasticsearch.index.shards.primaries` value in metricbeat
https://github.com/elastic/beats/pull/30408
- [ ] [update ECS mappings ](https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/core/src/main/resources/monitoring-kibana-mb.json)to add an alias for `kibana_stats.kibana.version` for `fetchKibanaVersions`
https://github.com/elastic/elasticsearch/pull/83930

Backporting to 8.0.1 notes:
- `createDatasetFilter` doesn't exist in 8.0.0 so will need add the `metricset.name` in a `bool` `should` along with `type` (legacy).  no need to add `data_stream.dataset` term
- no unit tests for each fetch in 8.0.0

Commit:
update queries to use metricset.name and fix paths

Commit:
update queries to use metricset.name and fix paths

kevinlog committed:
[Security Solution] Addition advanced policy options for 8.1 (#125230)

* [Security Solution] Addition advanced policy options for 8.1

* Update x-pack/plugins/security_solution/public/management/pages/policy/models/advanced_policy_schema.ts

Co-authored-by: Gabriel Landau <42078554+gabriellandau@users.noreply.github.com>

* Update x-pack/plugins/security_solution/public/management/pages/policy/models/advanced_policy_schema.ts

Co-authored-by: Gabriel Landau <42078554+gabriellandau@users.noreply.github.com>

* lint

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Gabriel Landau <42078554+gabriellandau@users.noreply.github.com>

dominiqueclarke committed:
uptime - adjust synthetics service sync errors behavior (#125169)

Commit:
Bump backport 7.0.1 (#125252)

gchaps committed:
[DOCS] Updates the Upgrade docs (#125288)

* [DOCS] Adds redirect for Standard Upgrade doc

* [DOCS] Updates Upgrade docs

* [DOCS] Incorporates review comments

* [DOCS] Adds info on saved objects

* [DOCS] Fixes broken link

* [DOCS] Fixes placement of section on unknown saved objects

* [DOCS] Removes duplicate topic

tylersmalley committed:
[kbn-pm] Add trailing newline in package.json (#125320)

flash1293 committed:
retry test (#125284)

fkanout committed:
[RAC][LOGS] Add reason message as a rule variable.  (#124978)

* Move types to the common folder and refactor code

* Add reason message as rule variable

* Revert "Move types to the common folder and refactor code"

This reverts commit 11692e7e1a9c56a64abb5824f540f26e0973cd38.

* Fix tests

* Fix tests

* Fix processGroupByRatioResults reason message

* Fix tests

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

shahzad31 committed:
[Observability overview] Fix overview page duplicate requests (#125204)

dominiqueclarke committed:
[Uptime] Convert ui monitor management combobox to checkboxes (#124917)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

emilioalvap committed:
[Uptime] Fix flaky monitor details tests (#124700)

* Rewrite flaky tests to e2e pipeline, add data initialization helpers, add monitor details page object

dokmic committed:
[Screenshotting] Add custom context support in the screenshot mode (#124942)

* Add custom context support to the screenshot mode plugin
* Delegate custom context handling to the screenshot mode plugin
* Move screenshot layout flag to the custom context

jsanz committed:
Fix index pattern reference in tooltip (#125220)

jonathan-buttner committed:
[Response Ops][Cases] Fetch alerts within observability (#123883)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

shahzad31 committed:
[Uptime] Added confirm modal (#125221)

cnasikas committed:
[Cases] Deprecate endpoints (#124773)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Kuznietsov committed:
[Lens] Fixed the order behavior of the visualization dimensions. (#125336)

* Added right sorting for pie visualizations.

VladLasitsa committed:
Add auto scale functionality to metric unified renderer (#124811)

* Add AutoSize component for metric

* Fix Checks

* moved auto scale from class to hoc component

* Remove unnecessary code

* Fix checks

* Refactoring

* Update shapshots

* Update snapshots

* Fix styles and change file name from auto_scale to with_auto_scale

* Fix some remarks

* Fix some remarks

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

vadimkibana committed:
improve variable name in test (#124849)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

alisonelizabeth committed:
[Upgrade Assistant] Update logic for handling reindex failures (#125246)

alisonelizabeth committed:
[Upgrade Assistant] Add warning when remote clusters are configured (#125138) (#125296)

shahzad31 committed:
[Uptime Monitor management] Update test configs (#125358)

hop-dev committed:
[Fleet] Tell users when assets are installed in a different space (#125278)

* Show new message when assets installed in separate space

* move variable into useEffect

juliaElastic committed:
fixed bug where fleet server instructions were incorrectly shown (#125357)

flash1293 committed:
[Lens] Version-aware sorting to data table (#125361)

nchaulet committed:
[Fleet] Make package policy name unique when copied multiple time (#125301)

stephmilovic committed:
[Security Solution] Data view override for sourcerer (#125116)

flash1293 committed:
[Lens] Fix focus on submitting filter popover (#125247)

* fix focus

* fix and make consistent

* fix test

dmlemeshko committed:
Scalability testing: Run Kibana with APM enabled (#125372)

* update config to enable apm

* wait 60 sec to get apm traces

* use suggested apm config

* set ELASTIC_APM_MAX_QUEUE_SIZE to collect full traces

* send APM data to Kibana stats

* increase delay shutdown

gbamparop committed:
Adds API tests for APM agent keys (#125302)

* Add API tests for apm agent keys

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

machadoum committed:
Bootstrap user page (#124701)

* Bootstrap user page

* Bootstrap user details page

* Delete ueba

* Create User detail flyout

* Add cypress test to User page

* Add Sourcerer to the users page

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

tsullivan committed:
[Reporting/Docs] Add clarity to reporting concepts (#125175)

* discourage setting enabled:false

* mark xpack.reporting.kibanaServer as experimental

* move kibanaServer section down

* add kibana privileges with reporting options screenshot

* rephrase why reporting.roles is depreccated

* screenshot for how to define a custom role the grants reporting privileges

* Apply suggestions from code review

Co-authored-by: Kaarina Tungseth <kaarina.tungseth@elastic.co>

* Apply suggestions from code review

Co-authored-by: Kaarina Tungseth <kaarina.tungseth@elastic.co>

* Update docs/settings/reporting-settings.asciidoc

Co-authored-by: Kaarina Tungseth <kaarina.tungseth@elastic.co>

* Apply suggestions from code review

Co-authored-by: Kaarina Tungseth <kaarina.tungseth@elastic.co>

* Update docs/settings/reporting-settings.asciidoc

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Kaarina Tungseth <kaarina.tungseth@elastic.co>

tsullivan committed:
[Reporting] Additional CSV job output metrics (#124600)

* [Reporting] Add number of rows in CSV as a job metric

* fix changed snapshot

jonathan-buttner committed:
[Response Ops][Cases] Add more granular checks before decoding user actions (#124896)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Christos Nasikas <christos.nasikas@elastic.co>

dominiqueclarke committed:
[Uptime] Monitor Details - fallback to saved objects when monitor details are unavailable from last ping (#125333)

* uptime - synthetic service - fallback to saved objects when monitor details cannot be found

* fix monitor url

* fix tests

darnautov committed:
[ML] Fix Trained Model stats and pipelines tab  (#125382)

stephmilovic committed:
[Security Solution] Use search strategy error in timeline (#125178)

jportner committed:
Add origin conflict checks when resolving import errors (#125241)

vitaliidm committed:
[Security solution][Detections] Fixes multiple Bulk Edit issues/suggestions and increases test coverage (#124525)

Issue: https://github.com/elastic/kibana/issues/125223
## Summary

- removes try/catch for bulk edit operation
- removes isElasticRule in bulk route API, replaces with rule.params.immutable check(as isElasticRule is used within telemetry)
- adds Cypress tests
- fixes case when index pattern action applied to ML rule. As ML rules don't have index pattern, so we will throw error if there is an attempt to run this action. It partially addresses https://github.com/elastic/kibana/issues/124918
- now checks if updated index patterns array is empty, if it is: throws error(https://github.com/elastic/kibana/issues/125223)

mattkime committed:
update docs (#125419)

Commit:
fix ccr query

Commit:
fix ccr query

neptunian commented on x-pack/plugins/monitoring/server/lib/alerts/fetch_ccr_read_exceptions.test.ts:
```diff
@@ -47,9 +47,32 @@ describe('fetchCCReadExceptions', () => {
+                bool: {
+                  should: [
+                    {
+                      nested: {
```
since nested fields can't be aliased when they don't belong to the same scope, these paths would throw an error when one mapping doesnt exist, so ignoring

klacabane replied:
That's how the logstash queries were designed as well so we're good as far as consistency goes :)

neptunian commented on x-pack/plugins/monitoring/server/lib/alerts/fetch_index_shard_size.test.ts:
```diff
@@ -187,6 +188,7 @@ describe('fetchIndexShardSize', () => {
                           '_index',
                           'index_stats.shards.primaries',
```
should the ecs equivalent of `index_stats.shards.primaries` be  `elasticsearch.index.shards.total` or `elasticsearch.cluster.stats.indices.shards.primaries`?

klacabane replied:
So the property should be `elasticsearch.index.shards.primaries` but it looks like it is [omitted when serialized to json](https://github.com/elastic/beats/blob/main/metricbeat/module/elasticsearch/index/data.go#L116-L128) in the 8.x metricbeat version. Looking at the < 8 version, these [properties are serialized as expected](https://github.com/elastic/beats/blob/7.16/metricbeat/module/elasticsearch/index/data_xpack.go#L108-L120)

klacabane replied:
I rebuilt metricbeat replacing the `-` with expected property names and they're now surfacing. These properties are not mapped, do we need them to be mapped for this query to work ?

<img width="482" alt="Screenshot 2022-02-15 at 18 30 12" src="https://user-images.githubusercontent.com/5239883/154116526-2d1d610b-abf8-4fd7-92a4-7be6ce085179.png">

neptunian replied:
Thanks Kevin!  No, I don't think it must be mapped as its referenced in `_source`.  I'll add the path.

FrankHassanabad committed:
Reduces timeouts from 10 minutes per test down to 2 minutes and changes one log.error to log.debug to avoid spamming cloud servers (#125432)

## Summary

Fixes https://github.com/elastic/kibana/issues/125319

Reduces timeouts from 10 minutes per test down to 2 minutes and changes log.error to log.debug to avoid spamming

I test this by making a failing test here:
```
x-pack/test/detection_engine_api_integration/security_and_spaces/tests/aliases.ts
```

By increasing `await waitForSignalsToBePresent(supertest, log, 4, [id]);` from `4` to `5`....Then I watched the logs and timed it to ensure it doesn't take 2 minutes.

I did this by running the server:
```sh
 node scripts/functional_tests_server.js --config test/detection_engine_api_integration/security_and_spaces/config.ts
```

And then running the client:
```sh
 node scripts/functional_test_runner.js --config test/detection_engine_api_integration/security_and_spaces/config.ts --include test/detection_engine_api_integration/security_and_spaces/tests/aliases.ts
```

If we start to see flake again on the regular build servers we might have to slightly increase this number again. If this doesn't allow failing servers to fully complete we might need to also decrease this number further or we might need to make this more configurable.

### Checklist

Delete any items that are not applicable to this PR.

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

mistic committed:
chore(NA): upgrade bazel into v5 (#125332)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Commit:
Fix Controls storybook config (#125305)

mattkime committed:
fix data view runtime fields api service key (#125430)

Heenawter committed:
[Dashboard] Transfer state when drilldown is opened in a new tab (#124770)

* Translate dashboard state to URL conditionally

* Add functional tests

* Fix typo in functional test descriptions

* Remove deprecated references

* Rename useUrl to be more specific

Commit:
fix cluster health query paths

Commit:
fix cluster health query paths

Commit:
update elasticsearch version mismatch

Commit:
update elasticsearch version mismatch

Commit:
update fetchLicense paths

Commit:
update fetchLicense paths

joeypoon committed:
[Security Solution] add feature usage notifications (#124766)

mistic committed:
chore(NA): upgrades rules nodejs to v4.0.0 (#125163)

* chore(NA): upgrade rules_node_js into v4.4.3

* chore(NA): update bazel typescript dep to 4.6.1

* chore(NA): downgrade rules nodejs to v4.0.0

* chore(NA): update dep on bazel/typescript pkg

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

pgayvallet committed:
Elasticsearch client: no longer default to using `meta: true` (#124488)

* Use `Client` interface instead of `KibanaClient`

* get rid of getKibanaEsClient and convertToKibanaClient

* get rid of last KibanaClient usages

* update usages and types in @kbn/securitysolution-es-utils

* fix some violations

* add sugar method around client mock

* update SO repository calls

* adapt more core usages

* export mock types

* batch 1

* batch 2

* batch 3

* batch 4

* batch 5

* batch 6

* batch 7

* batch 8

* batch 9

* security - batch 1

* security - batch 2

* security - batch 3

* last batch of initial violations

* fix resolve_time_pattern

* update generated doc

* fix /internal/index-pattern-management/preview_scripted_field endpoint

* fix monitoring's getLegacyClusterShim

* fix /api/snapshot_restore/privileges route

* fix UptimeESClient

* fix transforms/_nodes endpoint

* lint

* unit test fix - batch 1

* unit test fix - batch 2

* unit test fix - batch 3

* integration test fix - batch 1

* lint

* adapt ML client

* unit test fix - batch 4

* fix uptime test helper

* fix /api/transform/transforms/{transformId}/_update route

* fix ES client FTR test

* fix uptime unit test

* fix type errors on last unit tests

* fix RollupSearchStrategy call

* fix /internal/security/fields/{query} route

* fix GET /api/index_lifecycle_management/policies route

* fix mlClient.getDataFrameAnalytics

* fix APMEventClient

* fix security solution getBootstrapIndexExists

* fix data_enhanced's getSearchStatus

* remove unused @ts-expect-error

* fix unit tests due to latest code changes

* fix more calls in security_solution routes

* fix more calls in ml routes

* fix POST /api/index_management/component_templates route

* fix unit tests due to latest changes

* fix rule_registry's ResourceInstaller.createOrUpdateIndexTemplate

* fix more fleet client calls

* fix UA's GET cloud_backup_status route

* fix createLifecycleExecutorApiTest

* fix hasFleetServers

* fix unit tests due to latest changes

* changes due to last merge

* fix ml modelProvider.getModelsPipelines

* fix security_solution LifecycleQuery.search

* fix new CoreUsageDataService usage

* fix security solution's StatsQuery.search

* improve ml FTR assertions

* fix security_solution's EventsQuery.search

* fix EsClient type as we're keeping transport

* NITs

* clean RepositoryEsClient type

* update generated doc

* review comments

* adapt mlClient.anomalySearch signature

* remove unnecessary .then((body) => body)

* nit

* add unit tests for the client mocking functions

* fix new upgrade assistant /remote_clusters endpoint

patrykkopycinski committed:
[Osquery] Add telemetry for packs and saved queries (#122501)

mbondyra committed:
[Lens] small visual flyouts adjustments (#124970)

* [Lens] adjust visual changes for configuration flyouts

* paddings for add layer

* fix bug for extra space for reference lines

* fix non loading icon

* moving options for references

* remove warning from test

* show px inside the line thickness slider

* Update x-pack/plugins/lens/public/xy_visualization/visualization.tsx

Co-authored-by: Michael Marcialis <michael@marcial.is>

* refactor contextMenu

* Update add_layer.tsx

* change name, don't display decoration position if there's no decoration,

* left, auto, right & bottom, auto, top

* display none no

* icon value name changes

* tests

* icon in prepend

* cleaning up after the last piece of feedback

Co-authored-by: Michael Marcialis <michael@marcial.is>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

VladLasitsa committed:
[TSVB] Fix number fields which show NaN on invalid values (#124508)

* Replace input to EuiFieldNumber, fix createNumberHandler in case when value can be optional

* Add placeholder

* Fix remarks

* Fix name

* Replace `input` to `EuiFieldNumber` in window field

* Rename `isOptional` to `isClearable`

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

szabosteve committed:
[DOCS] Formats and fine-tunes ES query rule type docs (#125104)

vadimkibana committed:
Make embeddable example plugin not fail for read-only user (#125459)

* wrap write operation in try-catch block

* add space

darnautov committed:
[ML] Fix auto-refresh interval  (#124851)

* fix auto refresh

* update unit tests

* update mock

* show warning only with enabled auto refresh interval

* add extra assertion

* change default refresh interval to 10s

* handle time range updates, show warning with reset button

* fix side nav URLs update

* fix i18n

* update jest tests

* fix resolving of an active route

* change text, set default to 5s

* sync refresh interval, restore from timefiler on the AD list page

* fix smv refresh

* fix AE refresh

* fix setting of default value

* update jest test

* change toast life to 30s

* fix setting a global filter on init

juliaElastic committed:
updated fleet dependencies in readme (#125477)

gbamparop committed:
Add stories for agent keys (#125407)

* Add story for agent keys table

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

flash1293 committed:
fix formatting logic for terms (#125408)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

flash1293 committed:
[TSVB] Use panel interval for annotations (#125222)

renovate[bot] committed:
Update dependency elastic-apm-node to ^3.29.0 (#125453)

Co-authored-by: Renovate Bot <bot@renovateapp.com>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

yuliacech committed:
[Snapshot & Restore] Fix date parsing in the search bar (#125234)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

vitaliidm committed:
Fixes rules bulk edit tagd helper text (#125482)

Fixes helper text for tags combobox in rules bulk edit from:

## Before
<img width="1290" alt="Screenshot 2022-02-14 at 10 42 43" src="https://user-images.githubusercontent.com/92328789/153849260-dfd7d90c-e6ec-46cf-ab99-57396c12e20d.png">



## After
<img width="1281" alt="Screenshot 2022-02-14 at 10 42 08" src="https://user-images.githubusercontent.com/92328789/153849147-fb069a32-f0bb-4801-9191-77bcc9bc64dc.png">

joshdover committed:
[Fleet] Use registry version check on main (#125495)

hop-dev committed:
[Fleet] Trim whitespace from package policy names (#125400)

* Do not allow package policies with whitespace in their name

* Update x-pack/test/fleet_api_integration/apis/package_policy/create.ts

Co-authored-by: Nicolas Chaulet <n.chaulet@gmail.com>

Co-authored-by: Nicolas Chaulet <n.chaulet@gmail.com>

kpatticha committed:
[APM] Migrate service overview api test to apmApiClient (#125404)

* [APM] Migrate service overview api test to apmApiClient

* Pass path values via path property

Kuznietsov committed:
[PieVis] PartitionVis integration to Lens. (#123937)

* Removed pie/donut/mosaic/treemap expressions from lens

* Replaced pie/donut/mosaic/treemap expressions with expressions from expression_partition_vis

* Fixed bug with __other__ labels.

* Cleaned up not used fields at Lens.

* Added support of empty results for multiple chart types.]

* Refactored visualization_noresults.

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

mistic committed:
skip flaky suite (#123372)

nchaulet committed:
[Fleet] Fix invalid permission entry if the elastic agent package is not installed (#125447)

tylersmalley committed:
[ci] Splits Jest integration tests (#125454)

ppisljar committed:
fixing issue with expressions pending (#125520)

pheyos committed:
[ML] Docs screenshots - fix weblogs anomaly explorer screenshot (#125501)

This PR fixes the weblogs anomaly explorer docs screenshot by expanding the cell selection to the last element of the array and using another scrollIntoView.

darnautov committed:
[ML] Fix flaky Trained Models tests  (#125484)

* add extra assertion

* prevent extra call on init

* fix stats call

juliaElastic committed:
[Fleet] refresh policies on create and flyout open (#125499)

* refresh policies on create and flyout open

* fix checks

* fixed scenario when newly created policy should be preselected

tomsonpl committed:
[Osquery] Fix infinite loop (#124332)

Co-authored-by: Patryk Kopycinski <contact@patrykkopycinski.com>

stratoula committed:
[TSVB][Lens] Navigate to Lens with your current configuration (#114794)

* [Lens][TSVB] Convert to Lens

* Add logic for multiple series

* Basic formula

* Fix circular dependencies

* Minor cleanup

* Fix types

* fix jest tests

* Fix test

* Change the schema, add more styling options, fix bugs

* Supports time shift and custom date interval

* Fix types

* Fix some types

* Move edit in lens button to top nav menu

* Cleanup

* Further cleanup

* Add try it badge in menu, controlled by localStorage

* Add go back to app button

* Discard changes modal and go back to TSVB

* Update by value and by reference visualizations, delete existing by ref

* Fix bug

* Apply some changes

* get title and description only if has context

* Pass originating app, title and description from the savedVis

* By ref TSVB to by ref Lens

* Match TSVB cardinality with Lens unique_count function

* Support moving average

* Fix test

* Support derivative

* Support cumulative_sum

* Add overall functions

* Support filter ratio

* Refactor code for easier testing

* Fix bug with auto interval

* Fetch types from visualizations plugin

* Pipeline aggs compatible with percentile

* Add some bugs

* Support nesred aggs

* Mini refactor and support all aggregations to Math

* Transfer terms sorting options

* Transfer axis position

* Fix translations keys

* Revert

* Fix redirectToOrigin buttion when the there is no embeddableId but comes from dashboard

* Improve context identification

* Support yExtents

* Fix bug in formula caused by changes in the main branch

* Support formatters

* Support custom label

* Cleaning up

* Fix terms bugs

* Support filter breakdown by

* Fixes math bug and escapes filter ratio query

* Add some unit tests

* Testing triggerOptions payload

* Fix console warning

* Add more unit tests on TSVB function helpers

* Adds a unit test on the vis top nav menu testing the new menu item

* Add unit tests

* Fix unsupported palette bug, clean up, add a unit test case

* Add final unit tests

* Support timeScale in derivative

* Add functional tests

* Cleanup

* Fix jest test

* Fix some bugs

* Fix some math agg bugs

* Fix more bugs

* Fixes jest test

* Fix the problem with the dashboard state

* Hides the badge and link instead of disabling it

* Changes the text

* Adds menu item vertical separator

* Enhance the appLeace confirm modal to change the confirm button text and color

* Fixes CI

* Adress code review comments

* Address some of the comments

* Fix more bugs

* Fix more bugs

* Zero decimals for formatting

* fix tests

* Navigate from dashboard to TSVB to Lens hides the appLeave modal

* Adds support for terms on a date field

* Support filter by

* Move the trigger to the visualizations plugin

* Minor

* Fix CI

* Support percentage charts

* Improve the vertical separator

* Fixes on the appLeave logic

* Remove unecessary import

* Add badge to the nav item level

* Fix jest test

* Fi filter ratio and filter by bug

* Replace all occurences of a variable

* Nest badge into the button level

* Design improvements

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

dominiqueclarke committed:
[Uptime] handle null duration on ping list (#125438)

* uptime - handle null duration

* adjust types

* Update x-pack/plugins/uptime/server/lib/synthetics_service/service_api_client.ts

* raise per page of synthetics monitors

* Remove tooltip

gchaps committed:
[DOCS] Minor tweaks to upgrade docs (#125538)

KOTungseth committed:
[DOCS] Updates Upgrade Assistant API status (#125410)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

brianseeders committed:
[babel] Bump babel packages (#125446)

brianseeders committed:
[ci] Retry failed steps in on-merge pipeline (#125550)

dominiqueclarke committed:
synthetics e2e - use custom location when defined (#125560)

melissaburpo committed:
Update osquery.asciidoc to address doc issue (#125425)

This update fixes the issue raised in #125355

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

nchaulet committed:
[Fleet] Add retries when starting docker registry in integration tests (#125530)

vadimkibana committed:
[ML] Use Discover locator in Data Visualizer instead of URL Generator (#124283)

* use discover locator in data_visualizer to generate url

* pass in discover plugin into context

* format according to linter

* add console messages

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Commit:
[ci-stats] add Client class for accessing test group stats (#125164)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

pgayvallet committed:
SavedObjects management: change index patterns labels to data views (#125356)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

miltonhultgren committed:
[Logs UI] Support switching between log source modes (#124929)

* [Logs UI] Enable switching between log source modes (#120082)

* [Logs UI] Improve Data view selection section  (#124251)

* Update translation files

* Apply review requests

Apply useCallback optimization
Add telemetry tracking for users opting out of data view usage
Remove extra visual space at the bottom of the card
Improve initial render state of data view panel

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

mibragimov committed:
[Console] Support suggesting index templates v2 (#124655)

* Add autocomplete suggestions for index template api

* Add support for component templates

* Add unit tests

Co-authored-by: Muhammad Ibragimov <muhammad.ibragimov@elastic.co>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

shahzad31 committed:
[Uptime monitor management] Make index template installation retry (#125537)

janmonschke committed:
[SecuritySolution] Enrich threshold data from correct fields (#125376)

* fix: enrich threshold data from fields data

* test: add tests for field edge-cases

* test: test cases where value fields are missing

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

dgieselaar committed:
[APM] Lint rule for explicit return types (#124771)

dokmic committed:
[Reporting] Add additional PNG and PDF metrics  (#125450)

* Update browser driver to return metrics along with the results
* Add metrics to the reporting job
* Add metrics to the event logging
* Add screenshot metrics to the report info panel

dej611 committed:
[Graph] Make graph edges easier to click (#124053)

* :bug: Make edge easier to click on graph visualization

* :white_check_mark: Fix tests for new implementation

* :bug: Fix functional test

* :bug: Fix more functional tests

* :bug: Fix tests

* :bug: Fix hover behaviour

* :camera_flash: Update snapshot

* :white_check_mark: Adjust fucntional objects for new changes

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

MiriamAparicio committed:
Service overview e2e test (#125359)

* Add e2e test for service overview

* add test for comparison type change

* PR review comments addressed

* improve time range test

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

renovate[bot] committed:
Update dependency broadcast-channel to ^4.10.0 (#124740)

Co-authored-by: Renovate Bot <bot@renovateapp.com>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

flash1293 committed:
[Lens] Do not allow rarity in some cases (#125523)

* do not allow rarity in some cases

* use new params to build label

nchaulet committed:
[Fleet] Update copy for integration server (#125592)

hop-dev committed:
[Fleet] Do not mutate package policy update object  (#125622)

* do not modify object property

* add unit test

Commit:
Reverse parent child context relationship (#125486)

* reverse parent child context relationship

* bad merge

* ts

* ts

* fix jest

* try unblocking flaky test

* doc

juliaElastic committed:
[Fleet] extracted hook, clean up unused prop (#125610)

* extracted hook, clean up unused prop

* fixed tests

* added tests for hook

mistic committed:
skip flaky suite (#123253)

joshdover committed:
[Fleet] Avoid breaking setup when compatible package is not available in registry (#125525)

clintandrewhall committed:
[kibana_react] Enable Storybook for all of kibana_react (#125589)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

ymao1 committed:
[Task Manager] Adding list of explicitly de-registered task types (#123963)

* Adding REMOVED_TYPES to task manager and only marking those types as unrecognized

* Adding unit tests

* Fixing functional test

* Throwing error when registering a removed task type

* Adding migration

* Adding functional tests

* Cleanup

* Adding disabled siem signals rule type

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

awahab07 committed:
[Uptime][Monitor Management] Use  push flyout to show Test Run Results (#125017) (uptime/issues/445)

* Make run-once test results appear in a push flyout. Fixup tooltip. Fixup action buttons order.

* Wrapping Monitor Fields form rows when Test Run flyout is open.

* Only show step duration trend if it's an already saved monitor. Stop showing "Failed to run steps" until Test Run steps are done loading.

uptime/issues/445

Co-authored-by: shahzad31 <shahzad.muhammad@elastic.co>

juliaElastic committed:
revert package policy validation that caused issue with input groups (#125657)

nchaulet committed:
[Fleet] Fix output form validation for trusted fingerprint (#125662)

vadimkibana committed:
Use Discover locator to generate URL (#124282)

* use Discover locator to generate URL

* improve locator check

* do not import discover plugin

* allow for share plugin to be missing

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

szabosteve committed:
[DOCS] Removes technical preview flag from DFA ML page. (#125497)

dasansol92 committed:
Remove meta field before create/update event filter item (#125624)

janmonschke committed:
[SecuritySolution][Threat Hunting] Fix a couple of field ids for highlighted fields (#124941)

* fix: use correct DNS field id

* fix: for behavior alerts we should display rule.description

cauemarcondes committed:
[APM] Bug: Service maps popover detail metrics are aggregates over all transaction types (#125580)

* fixing failure transaction rate on service maps

* addressing PR comments

dej611 committed:
[Lens][Example] Testing playground for Lens embeddables (#125061)

* :alembic: First pass

* :sparkles: Add Lens testing playground

* :sparkles: Add partition preset

* :ok_hand: Addressed feedback

* :lipstick: Add thumbnail for example catalog

* :wrench: Convert image to png

* :fire: Remove extra debug buttons

* :bug: Fix error handling

* :bug: Fix buttons bug

ashokaditya committed:
[SecuritySolution][Endpoint] Add a user with kibana_system and superuser role for adding fake data (#125476)

* add a user with kibana_system and superuser role for adding fake data

fixes elastic/security-team/issues/2908

* Check user exists before adding

review changes

* Delete endpoint_user afterwards

review changes

* user API response to simplify adding user

* simplify type

* allow picking username and password from cli

review suggestions

* do not add new user by default, require it only when fleet is enabled

review changes

* use URL

review changes

* update protocol with URL as well

kpatticha committed:
[APM] Reduce maxNumServices from 500 to 50 (#125646)

* [APM] Reduce maxNumServices from 500 to 50

* Fix snapshot

klacabane committed:
use ecs properties in fetch_index_shard_size

klacabane committed:
use ecs properties in fetch_index_shard_size

klacabane commented:
This shards alert will need https://github.com/elastic/beats/pull/30408 to be merged

madirey committed:
[Security Solution][Timeline] Use first event in threshold set for 'from' value in timeline (#123282)

* Use first event in threshold set for 'from' value in timeline

* lint

* Linting and test fix

* Tests

* Fix signal generation tests

* Update more 'from' dates in threshold tests

* More test fixes

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

madirey committed:
[Security Solution] Upgrade tests for DE rule types - alerts on legacy alerts (#125331)

* Add integration tests for alerts-on-legacy-alerts

* Remove query rule tests from prior location - they were moved

* Remove 'kibana' field from alerts on legacy alerts

* Fix tests

* Delete alerts before proceeding from compatibility tests

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

vadimkibana committed:
use locator in osquery plugin (#125698)

jbudz committed:
[cft] Generate update plan based on current configuration (#125317)

When a Cloud deployment is updated, we're reusing the original creation
template.  In cases where settings are manually overridden. these changes will be
lost when a deployment is updated.  Instead of using the base template,
this queries the cloud API for an update payload as a base.

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

jbudz committed:
[ci] Verify docker contexts (#122897)

* [ci] Verify docker contexts

* bootstrap

* debug

* mkdir target

* change subdomain if snapshot

* move to separate pipeline

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

FrankHassanabad committed:
[Security Solutions] Updates usage collector telemetry to use PIT (Point in Time) and restructuring of folders (#124912)

## Summary

Changes the usage collector telemetry within security solutions to use PIT (Point in Time) and a few other bug fixes and restructuring.

* The main goal is to change the full queries for up to 10k items to be instead using 1k batched items at a time and PIT (Point in Time). See [this ticket](https://github.com/elastic/kibana/issues/93770) for more information and [here](https://github.com/elastic/kibana/pull/99031) for an example where they changed there code to use 1k batched items. I use PIT with SO object API, searches, and then composite aggregations which all support the PIT. The PIT timeouts are all set to 5 minutes and all the maximums of 10k to not increase memory more is still there. However, we should be able to increase the 10k limit at this point if we wanted to for usage collector to count beyond the 10k. The initial 10k was an elastic limitation that PIT now avoids.
* This also fixes a bug where the aggregations were only returning the top 10 items instead of the full 10k. That is changed to use [composite aggregations](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-composite-aggregation.html). 
* This restructuring the folder structure to try and do [reductionism](https://en.wikipedia.org/wiki/Reductionism#In_computer_science) best we can. I could not do reductionism with the schema as the tooling does not allow it. But the rest is self-repeating in the way hopefully developers expect it to be. And also make it easier for developers to add new telemetry usage collector counters in the same fashion.
* This exchanges the hand spun TypeScript types in favor of using the `caseComments` and the `Sanitized Alerts` and the `ML job types` using Partial and other TypeScript tricks.
* This removes the [Cyclomatic Complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity) warnings coming from the linters by breaking down the functions into smaller units.
* This removes the "as casts" in all but 1 area which can lead to subtle TypeScript problems.
* This pushes down the logger and uses the logger to report errors and some debug information

### Checklist

Delete any items that are not applicable to this PR.

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

alvarezmelissa87 committed:
[ML] Data Frame Analytics saved search creation functional tests (#125040)

* create filter saved search test for each dfa job type

* add search query test cases

* temp comment of failing with insuff memory test case

* remove problem case and update test text

* Use downsampled farequote archive

* remove unnecessary file

Co-authored-by: Robert Oskamp <robert.oskamp@elastic.co>

alvarezmelissa87 committed:
[ML] Data Frame Analytics audit messages api test (#125325)

* wip: add beginning of messages api test

* adds dfa audit messages api test

* use retry instead of running job

joeypoon committed:
[Security Solution] unskip tests (#125675)

neptunian commented on x-pack/plugins/monitoring/server/lib/alerts/fetch_index_shard_size.ts:
```diff
@@ -133,15 +135,20 @@ export async function fetchIndexShardSize(
       const {
         _index: monitoringIndexName,
-        _source: { index_stats: indexStats },
+        _source: { index_stats: indexStats, elasticsearch },
```
if you just declare `indexStats` like this:
`const indexStats = topHit._source.index_stats || topHit._source.elasticsearch?.index;`, I don't think you'd need to make any other changes @klacabane

janmonschke committed:
fix: add rule type to global fields for the alert flyout (#125676)

jbudz committed:
Revert "[build] Include x-pack example plugins when using example-plugins flag (#120697)" (#125729)

This reverts commit 3a4c6030b901241eb8e9fbef337c5d36f6fd2fb5.

Commit:
[App Search] New flyout to start a crawl with custom settings (#124999)

FrankHassanabad committed:
[Security Solutions] Exposes the search_after and point in time (pit) from saved objects to exception lists (#125182)

## Summary

Exposes the functionality of
* search_after
* point in time (pit)

From saved objects to the exception lists. This _DOES NOT_ expose these to the REST API just yet. Rather this exposes it at the API level to start with and changes code that had hard limits of 10k and other limited loops. I use the batching of 1k for this at a time as I thought that would be a decent batch guess and I see other parts of the code changed to it. It's easy to change the 1k if we find we need to throttle back more as we get feedback from others.

See this PR where `PIT` and `search_after` were first introduced: https://github.com/elastic/kibana/pull/89915
See these 2 issues where we should be using more paging and PIT (Point in Time) with search_after: https://github.com/elastic/kibana/issues/93770 https://github.com/elastic/kibana/issues/103944

The new methods added to the `exception_list_client.ts` client class are:
* openPointInTime
* closePointInTime
* findExceptionListItemPointInTimeFinder
* findExceptionListPointInTimeFinder
* findExceptionListsItemPointInTimeFinder
* findValueListExceptionListItemsPointInTimeFinder

The areas of functionality that have been changed:
* Exception list exports
* Deletion of lists
* Getting exception list items when generating signals

Note that currently we use our own ways of looping over the saved objects which you can see in the codebase such as this older way below which does work but had a limitation of 10k against saved objects and did not do point in time (PIT)

Older way example (deprecated):
```ts
  let page = 1;
  let ids: string[] = [];
  let foundExceptionListItems = await findExceptionListItem({
    filter: undefined,
    listId,
    namespaceType,
    page,
    perPage: PER_PAGE,
    pit: undefined,
    savedObjectsClient,
    searchAfter: undefined,
    sortField: 'tie_breaker_id',
    sortOrder: 'desc',
  });
  while (foundExceptionListItems != null && foundExceptionListItems.data.length > 0) {
    ids = [
      ...ids,
      ...foundExceptionListItems.data.map((exceptionListItem) => exceptionListItem.id),
    ];
    page += 1;
    foundExceptionListItems = await findExceptionListItem({
      filter: undefined,
      listId,
      namespaceType,
      page,
      perPage: PER_PAGE,
      pit: undefined,
      savedObjectsClient,
      searchAfter: undefined,
      sortField: 'tie_breaker_id',
      sortOrder: 'desc',
    });
  }
  return ids;
```

But now that is replaced with this newer way using PIT:
```ts
  // Stream the results from the Point In Time (PIT) finder into this array
  let ids: string[] = [];
  const executeFunctionOnStream = (response: FoundExceptionListItemSchema): void => {
    const responseIds = response.data.map((exceptionListItem) => exceptionListItem.id);
    ids = [...ids, ...responseIds];
  };

  await findExceptionListItemPointInTimeFinder({
    executeFunctionOnStream,
    filter: undefined,
    listId,
    maxSize: undefined, // NOTE: This is unbounded when it is "undefined"
    namespaceType,
    perPage: 1_000,
    savedObjectsClient,
    sortField: 'tie_breaker_id',
    sortOrder: 'desc',
  });
  return ids;
```

We also have areas of code that has perPage listed at 10k or a constant that represents 10k which this removes in most areas (but not all areas):
```ts
      const items = await client.findExceptionListsItem({
        listId: listIds,
        namespaceType: namespaceTypes,
        page: 1,
        pit: undefined,
        perPage: MAX_EXCEPTION_LIST_SIZE, // <--- Really bad to send in 10k per page at a time
        searchAfter: undefined,
        filter: [],
        sortOrder: undefined,
        sortField: undefined,
      });
```

That is now:
```ts
      // Stream the results from the Point In Time (PIT) finder into this array
      let items: ExceptionListItemSchema[] = [];
      const executeFunctionOnStream = (response: FoundExceptionListItemSchema): void => {
        items = [...items, ...response.data];
      };

      await client.findExceptionListsItemPointInTimeFinder({
        executeFunctionOnStream,
        listId: listIds,
        namespaceType: namespaceTypes,
        perPage: 1_000,
        filter: [],
        maxSize: undefined, // NOTE: This is unbounded when it is "undefined"
        sortOrder: undefined,
        sortField: undefined,
      });
```

Left over areas will be handled in separate PR's because they are in other people's code ownership areas.

### Checklist
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

maksimkovalev committed:
#121908 - [Maps] Use UI counters to instrument when a geo field is visualized with visualizeGeoFieldAction (#123540)

* #121908 - Maps usage tracking feature

* #e-121908 - refactoring; Added naming constants for used apps;

* #121908 - updates for UiCounterMetricType, schema, metric analytics

* #121908 - refactoring; Fixed import syntax, type in report

* #121908 - refactoring

* #121908 - refactoring

* 121908 - refactoring

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

TinaHeiligers committed:
Adds nav link to uisettings tutorial (#125736)

tylersmalley committed:
Updates to Jest configuration (#125727)

elasticmachine commented:
Pinging @elastic/infra-monitoring-ui (Team:Infra Monitoring UI)

jasonrhodes commented:
@klacabane / @neptunian are all of the changes I'm seeing enumerated in this ticket necessary to get alerts to start working for a user who has upgraded to 8.x, or are some of these fixes to related problems that they would run into? I want to make sure that we don't block the merge of this for anything except what's necessary to make sure alerts generally work in 8.x, because my understanding is that right now they are completely broken.

We should definitely wait on comprehensive testing of the CCR rule because that's not going to happen in time for 8.0.1. 

Is [this beats change](https://github.com/elastic/beats/pull/30408) necessary to unblock the main problem (missing "type", need to use `metricset.name` instead), or is it a related problem? 

And is [this ES change](https://github.com/elastic/elasticsearch/pull/83930) related to the missing type problem?

If things get in for 8.0.1 that fix related problems, that's great, but I want to focus in on what's absolutely necessary to unblock the biggest bug. 

Thanks for your help!

jasonrhodes committed:
Fixed type error for body vs response

I fixed a merge conflict locally but had rebased and didn't want to force
push so I pulled and commited the update here instead.

jasonrhodes commented:
Merge explosion, I'm closing this and will re-open a new PR.

jasonrhodes commented:
Replaced by https://github.com/elastic/kibana/pull/125748
